### PR TITLE
Added q_col and p_row methods and documentation for getting the latent factors of a column or row

### PR DESCRIPTION
**Pull request by guccialex**

### CODE REVIEW

File: src/model.rs [hunk 1]
```diff
@@ -61,6 +61,30 @@ impl Model {
         unsafe { std::slice::from_raw_parts((*self.model).q, (self.columns() * self.factors()) as usize) }
     }
 
+    pub fn p_row(&self, row_index: i32) -> Option<&[f32]>{
+        use std::convert::TryFrom;
+        if let Ok(factors) = usize::try_from( self.factors() ){
```
Comment (ankane) [inline src/model.rs]: What's the reasoning behind using `try_from`?

File: README.md [hunk 2]
```diff
@@ -37,13 +37,21 @@ Make predictions
...
+```
+
+Get all the latent factors (these approximate the training matrix)
 
 ```rust
 model.p_factors();
 model.q_factors();
 ```
 
+
```
Comment (ankane) [inline README.md]: Please remove / keep the changeset minimal.

**Review by ankane — COMMENTED** [review]
Thanks @guccialex! Left a few comments inline.

Also, I was thinking another way would be to create iterator functions (`p_iter()` and `q_iter()`) to make it efficient to iterate over all the rows / columns. Would that also work for your use case?

File: src/model.rs (hunk 1 above)
Comment (guccialex) [reply src/model.rs]: It was to not panic if there were negative factors. But that shouldn't happen, and it probably should panic if it does, so I've removed it.

File: README.md (hunk 2 above)
Comment (guccialex) [reply README.md]: It's been fixed and put into one commit

**Comment by guccialex** [comment]
For my purposes, I'm training 250_000 items with a factor size of 200, and reading all of the factors seemed quick enough using these methods, but let me get the exact time.

**Comment by guccialex** [comment]
Collecting them all, calling q_col ~250,000 times takes ~75 ms.
I don't think an iterator is needed. I can't think of when you'd want every factor in order, unless you were putting them in your own data structure, which time isn't too important for, and even if it was, this is fine for.

But I do have something like:
```
for index in 0..model.columns(){
    if let Some(factors) = model.q_col(index){
```
And that probably would be a common pattern using this library

**Comment by ankane** [comment]
Thanks @guccialex!

After thinking about it more, I think we can use `p` and `q` for the method names, so I updated in a follow-up commit.

Also, for the pattern above, added `p_iter` and `q_iter` methods in e276db1cece6b225dbe46a31a93fd2e6e3b7f1f6, so you can do:

```rust
for (index, factors) in model.p_iter().enumerate() {
  // code
}
```

**Pull request merged**